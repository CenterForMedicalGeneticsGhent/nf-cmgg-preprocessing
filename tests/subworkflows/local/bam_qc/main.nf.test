nextflow_workflow {

    name "Test Workflow BAM_QC"
    script "subworkflows/local/bam_qc/main.nf"
    workflow "BAM_QC"

    test("QC - No Picard") {

        when {
            params {
                disable_picard_metrics = true
            }
            workflow {
                """
                // ch_bam_bai
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true)
                ])

                // ch_fasta_fai
                input[1] = Channel.of([
                    [id:'genome'],
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true),
                    ])


                // ch_fasta_dict
                input[2] = Channel.of([
                    [id:'genome'],
                    file(params.test_data['sarscov2']['genome']['genome_dict'], checkIfExists: true),
                    ])

                // ch_target_interval
                input[3] = Channel.of( file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true))

                // ch_bait_interval
                input[4] = Channel.of( file(params.test_data['sarscov2']['genome']['baits_bed'], checkIfExists: true))
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("QC - Picard HS metrics") {

        when {
            params {
                disable_picard_metrics = false
            }
            workflow {
                """
                // ch_bam_bai
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true)
                ])

                // ch_fasta_fai
                input[1] = Channel.of([
                    [id:'genome'],
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true),
                    ])


                // ch_fasta_dict
                input[2] = Channel.of([
                    [id:'genome'],
                    file(params.test_data['sarscov2']['genome']['genome_dict'], checkIfExists: true),
                    ])

                // ch_target_interval
                input[3] = Channel.of( file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true))

                // ch_bait_interval
                input[4] = Channel.of( file(params.test_data['sarscov2']['genome']['baits_bed'], checkIfExists: true))
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("QC - Picard WGS metrics") {

        when {
            params {
                disable_picard_metrics = false
            }
            workflow {
                """
                // ch_bam_bai
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true)
                ])

                // ch_fasta_fai
                input[1] = Channel.of([
                    [id:'genome'],
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true),
                    ])


                // ch_fasta_dict
                input[2] = Channel.of([
                    [id:'genome'],
                    file(params.test_data['sarscov2']['genome']['genome_dict'], checkIfExists: true),
                    ])

                // ch_target_interval
                input[3] = Channel.of([])

                // ch_bait_interval
                input[4] = Channel.of([])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
