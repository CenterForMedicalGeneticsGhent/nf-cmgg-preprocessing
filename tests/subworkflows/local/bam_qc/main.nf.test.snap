{
    "QC - Picard WGS metrics": {
        "content": [
            {
                "versions": [
                    "versions.yml:md5,c679001d803dacd120a83e77ce7ebf7c",
                    "versions.yml:md5,3fe0fb0d490cf1bf4adef6054a996aaa",
                    "versions.yml:md5,807bb6c5f45f7f79747bf0ac76b4b1d3",
                    "versions.yml:md5,88ef3b5bbeff0dcac96fb5a6371d5b37"
                ],
                "metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.idxstats:md5,e179601fa7b8ebce81ac3765206f6c15",
                            "test.flagstat:md5,a53f3d26e2e9851f7d528442bbfe9781",
                            "test.stats:md5,e40e4071746db8b5eed5012c59b3304a",
                            "test.CollectMultipleMetrics.alignment_summary_metrics:md5,37140daed05c6055ad2ce374016b6380",
                            "test.CollectMultipleMetrics.base_distribution_by_cycle_metrics:md5,1d380d44c3e8850d8a6aa2b110e143f2",
                            "test.CollectMultipleMetrics.insert_size_metrics:md5,e1ec34b2728108568d3037674192c3e5",
                            "test.CollectMultipleMetrics.quality_by_cycle_metrics:md5,c00773b5b19ff34c1c0ac424305e1455",
                            "test.CollectMultipleMetrics.quality_distribution_metrics:md5,87ff3689c629545af9be5e8adea7dd9d"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2022-12-06T14:40:10+0000"
    }
}